cluster/cluster.yaml

```diff
@@ -2006,6 +2006,101 @@ Resources:
             Resource: "*"
             Action:
             - "kms:Decrypt"
+  IdentityProvider:
+    Type: Custom::IdentityProvider
+    Properties:
+      ServiceToken: !GetAtt IdentityProviderCreator.Arn
+      Region: !Ref "AWS::Region"
+      ProviderName: "{{ .Cluster.LocalID }}.{{ .Values.hosted_zone }}"
+  IdentityProviderCreator:
+    Type: AWS::Lambda::Function
+    Properties:
+      Runtime: python2.7
+      Handler: index.lambda_handler
+      MemorySize: 128
+      Role: !GetAtt LambdaExecutionRole.Arn
+      Timeout: 30
+      Code:
+        ZipFile: !Sub |
```

Comment:
nit: Maybe we could deploy using a container rather than a zip. WDYT?

Reply:
Do you mean we can use a docker image to hold the code instead on inlining it here?